packages/grid/x-data-grid-pro/src/components/GridRowReorderCell.tsx

```diff
@@ -41,11 +42,12 @@ const GridRowReorderCell = (params: GridRenderCellParams) => {
   // TODO: remove sortModel and treeDepth checks once row reorder is compatible
   const isDraggable = React.useMemo(
     () =>
-      !!rootProps.rowReordering &&
+      !!rowReordering &&
       !sortModel.length &&
       treeDepth === 1 &&
-      Object.keys(editRowsState).length === 0,
-    [rootProps.rowReordering, sortModel, treeDepth, editRowsState],
+      Object.keys(editRowsState).length === 0 &&
+      isRowDraggable(apiRef.current.getRowParams(params.id)),
```

Comment:
Based on the comments in #4754 I don't know what users expect. They may want to keep some rows always at the bottom, like pinned rows but not pinned, or just disable the icon. If we disable the icon, then `isRowDraggable` could be only a prop of `GridRowReorderCell`, not for the grid. I think we should also add a demo of this feature, to clarify https://github.com/mui/mui-x/issues/4754#issuecomment-1187580566

Reply:
Noted, I'll add a demo. We can also go with the approach of adding a prop on the `GridRowReorderCell`. I went with the current approach as it's the closest in relation to what we have for the columns.
Regarding the first point of having pinned rows - same as the columns. You can reorder around them but can't reorder the one that's disabled.